syntax = "proto3";

package idl_compiler.compiler.enumeration;

import "idl_compiler/compiler/shared.proto";

message Enumeration {
    idl_compiler.compiler.Type type = 1;

    string name = 2;

    string item_prefix = 3;

    repeated EnumerationItem items = 4;

    map<string, string> documentation = 5;
}

message EnumerationItem {
    string name = 1;

    string value = 2;

    map<string, string> documentation = 3;
}
